### DISCUSSION TITLE
Setting Classes on Child Elements from Parent Container in XAML

### DISCUSSION THREAD

**Discussion by hilk** [AvaloniaUI/Avalonia#13569] [Q&A] [answered]
I'm currently working with Avalonia 10.18 and I'm trying to find a way to set classes on child elements directly from their parent container in XAML. I have a Border with multiple TextBlock elements, and I want to assign a common class to all the child TextBlock elements without setting the Classes property individually.

I've tried using the following XAML syntax:
```
<Border>
    <Border.Styles>
        <Style Selector="TextBlock">
            <Setter Property="Classes" Value="TestStyle"/>
        </Style>
    </Border.Styles>
</Border>
```

However, I've encountered limitations, and it seems that the Classes property is not directly applicable to child elements from the parent container in this manner.

Is there a recommended way to achieve this without setting classes individually for each child element? 

Thanks!

**Comment by timunie** [comment, marked as answer]
That is not supported. You may want to write a Behavior or AttachedProperty for it.
